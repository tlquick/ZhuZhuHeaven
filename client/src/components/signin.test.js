import { BrowserRouter } from "react-router-dom";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Signin from "./signin";
const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));
afterEach(() => {
  cleanup();
});
test("renders signin page", () => {
  render(
    <BrowserRouter>
      <Signin />
    </BrowserRouter>
  );
  const signinElement = screen.getByTestId("signin-1");
  expect(signinElement).toBeInTheDocument();
  expect(signinElement).toContainHTML("img"); //contains a <img> tag
  expect(signinElement).toContainHTML("form"); //contains a <form> tag
});
test("matches signin snapshot", () => {
  const tree = renderer.create(<Signin />).toJSON();
  expect(tree).toMatchSnapshot();
});
