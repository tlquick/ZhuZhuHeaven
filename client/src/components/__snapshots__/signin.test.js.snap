// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches signin snapshot 1`] = `
<div>
  <div
    className="card card-container"
    data-testid="signin-1"
  >
    <img
      alt="profile-img"
      className="profile-img-card mx-auto"
      src="//ssl.gstatic.com/accounts/ui/avatar_2x.png"
    />
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <button
          className="sign-button btn btn-primary btn-lg btn-block"
        >
          <span>
            Sign in
          </span>
        </button>
      </div>
      <button
        style={
          Object {
            "display": "none",
          }
        }
      />
      <div
        className="link"
      >
        Not registered? 
        <a
          href="/user/signup"
        >
          Sign Up
        </a>
      </div>
    </form>
  </div>
</div>
`;
